ui/app/AppLayouts/Chat/components/ChannelIcon.qml

```diff
@@ -0,0 +1,79 @@
+import QtQuick 2.3
+import QtQuick.Controls 2.3
+import QtQuick.Layouts 1.3
+import Qt.labs.platform 1.1
+import "../../../../imports"
+import "../../../../shared"
+
+Rectangle {
+  property var topMargin: Theme.smallPadding
+  property var bottomMargin: Theme.smallPadding
+  property var channelName
+  property var channelType
+  id: contactImage
+  width: width ? width : 36
+  height: height ? height: 36
+  anchors.left: parent.left
+  anchors.leftMargin: Theme.padding
+  anchors.top: parent.top
+  anchors.topMargin: topMargin
+  anchors.bottom: parent.bottom
+  anchors.bottomMargin: bottomMargin
+  radius: 50
+
+  Loader {
+    sourceComponent: channelType == Constants.chatTypeOneToOne ? imageIdenticon : letterIdenticon
+    anchors.fill: parent
+  }
+
+  Component {
+    id: letterIdenticon
+    Rectangle {
+      width: width ? width : 36
+      height: height ? height: 36
+      radius: 50
+      color: {
+          const color = chatsModel.getChannelColor(channelName)
+          if (!color) {
+              return Theme.transparent
+          }
+          return color
+      }
+
+      Text {
+        text: {
+          if (channelType == Constants.chatTypeOneToOne) {
+            return channelName;
+          } else {
+            return (channelName.charAt(0) == "#" ? channelName.charAt(1) : channelName.charAt(0)).toUpperCase();
+          }
+        }
+        opacity: 0.7
+        font.weight: Font.Bold
+        font.pixelSize: 21
+        color: "white"
+        anchors.horizontalCenter: parent.horizontalCenter
+        anchors.verticalCenter: parent.verticalCenter
+      }
+    }
+  }
+
+  Component {
+    id: imageIdenticon
+    Rectangle {
+      width: 40
+      height: 40
```

Comment:
You set the width above to `width ? width : 36`, you probably need to do it here too (with the correct binding to the parent's width)

Reply:
Actually, the ternary one was a left over. I noticed that built-in properties can't be overriden.